lib/matplotlib/backends/backend_agg.py

```diff
@@ -147,24 +147,34 @@ def draw_path(self, gc, path, transform, rgbFace=None):
         """
         nmax = rcParams['agg.path.chunksize'] # here at least for testing
         npts = path.vertices.shape[0]
+        
         if (nmax > 100 and npts > nmax and path.should_simplify and
                 rgbFace is None and gc.get_hatch() is None):
-            nch = np.ceil(npts/float(nmax))
-            chsize = int(np.ceil(npts/nch))
+            nch = np.ceil(npts / float(nmax))
+            chsize = int(np.ceil(npts / nch))
             i0 = np.arange(0, npts, chsize)
             i1 = np.zeros_like(i0)
             i1[:-1] = i0[1:] - 1
             i1[-1] = npts
             for ii0, ii1 in zip(i0, i1):
-                v = path.vertices[ii0:ii1,:]
+                v = path.vertices[ii0:ii1, :]
                 c = path.codes
                 if c is not None:
                     c = c[ii0:ii1]
-                    c[0] = Path.MOVETO # move to end of last chunk
+                    c[0] = Path.MOVETO  # move to end of last chunk
                 p = Path(v, c)
-                self._renderer.draw_path(gc, p, transform, rgbFace)
+                try:
+                    self._renderer.draw_path(gc, p, transform, rgbFace)
+                except OverflowError as e:
+                    e.args = ("Exceeded cell block limit (set 'agg.path.chunksize' rcparam)",)
```

Comment:
in fact, if we raise new overflow error. We get something like this 

> Exception in Tkinter callback
> Traceback (most recent call last):
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/backends/backend_agg.py", line 171, in draw_path
>     self._renderer.draw_path(gc, path, transform, rgbFace)
> OverflowError: In draw_path: Exceeded cell block limit
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib_env/lib/python3.5/tkinter/__init__.py", line 1562, in __call__
>     return self.func(*args)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/backends/backend_tkagg.py", line 284, in resize
>     self.show()
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/backends/backend_tkagg.py", line 355, in draw
>     FigureCanvasAgg.draw(self)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/backends/backend_agg.py", line 475, in draw
>     self.figure.draw(self.renderer)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/artist.py", line 68, in draw_wrapper
>     return draw(artist, renderer, *args, **kwargs)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/figure.py", line 1240, in draw
>     renderer, self, artists, self.suppressComposite)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/image.py", line 139, in _draw_list_compositing_images
>     a.draw(renderer)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/artist.py", line 68, in draw_wrapper
>     return draw(artist, renderer, *args, **kwargs)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/axes/_base.py", line 2386, in draw
>     mimage._draw_list_compositing_images(renderer, self, artists)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/image.py", line 139, in _draw_list_compositing_images
>     a.draw(renderer)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/artist.py", line 68, in draw_wrapper
>     return draw(artist, renderer, *args, **kwargs)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/lines.py", line 800, in draw
>     line_func(renderer, gc, tpath, affine.frozen())
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/lines.py", line 1248, in _draw_solid
>     renderer.draw_path(gc, path, trans)
>   File "/home/herilalaina/Documents/contribution/matplotlib_contrib/matplotlib/lib/matplotlib/backends/backend_agg.py", line 173, in draw_path
>     raise OverflowError("Exceeded cell block limit (set 'agg.path.chunksize' rcparam)")
> OverflowError: Exceeded cell block limit (set 'agg.path.chunksize' rcparam)

I will make the change if It's okay.